File: src/tests/integration/Bc.Endpoint.Integration.Tests/GitHubPullRequestCreationSender.cs
using System;
using System.Threading.Tasks;
using Bc.Contracts.Internals.Endpoint.GitHubPullRequestCreation.Messages;
using NServiceBus;
using NUnit.Framework;

namespace Bc.Endpoint.Integration.Tests
{
    [TestFixture]
    [Ignore("Only for manual tests")]
    public class GitHubPullRequestCreationSender
    {
        private IEndpointInstance endpointInstance;

        [SetUp]
        public async Task SetUp()
        {
            this.endpointInstance = await EndpointFactory.GetSenderEndpoint().ConfigureAwait(false);
        }

        [TearDown]
        public Task TearDown()
        {
            return this.endpointInstance.Stop();
        }

        [Test]
        public async Task RequestCreateGitHubPullRequest_Send_NoException()
        {
            // Arrange
            var commentId = Guid.NewGuid();
            const string fileName = @"_posts/2018-05-27-test.md";
            const string content = "new_comment";
            var addedDate = DateTime.UtcNow;

            var message = new RequestCreateGitHubPullRequest(
                commentId,
                fileName,
                content,
                addedDate);

            // Act
            await this.endpointInstance.Send(message).ConfigureAwait(false);

            // Assert
            Assert.Pass();
        }
    }
}

File: src/tests/integration/Bc.Endpoint.Integration.Tests/GitHubPullRequestCreationSender.fs
module Bc.Endpoint.Integration.Tests.GitHubPullRequestCreationSender

open System
open System.Threading.Tasks
open Bc.Contracts.Internals.Endpoint.GitHubPullRequestCreation.Messages
open NServiceBus
open NUnit.Framework

[<Ignore("Only for manual tests")>]
module Sender =

    let sendMessage<'T> (message:'T) =
        async {
            let! endpoint = EndpointFactory.getSenderEndpoint () |> Async.AwaitTask
            do! endpoint.Send(message) |> Async.AwaitTask
            do! endpoint.Stop() |> Async.AwaitTask
        } |> Async.RunSynchronously

    [<Test>]
    let requestCreateGitHubPullRequest_send_noException () =

        // Arrange
        let commentId = Guid.NewGuid()
        let fileName = @"_posts/2018-05-27-test.md"
        let content = "new_comment"
        let addedDate = DateTime.UtcNow

        let message =
            RequestCreateGitHubPullRequest (
                commentId,
                fileName,
                content,
                addedDate
            )

        // Act
        sendMessage<RequestCreateGitHubPullRequest> message

        // Assert
        Assert.Pass()
